lesson-02-assignment.rb

```
@@ -0,0 +1,67 @@
+#QUESTION 1 
+Trimester.create(year: '2026', term: 'Spring', application_deadline: '2026-02-15', start_date: '2026-03-01', end_date: '2026-05-31')
+#TRANSACTION (0.0ms)  BEGIN immediate TRANSACTION /*application='DocRails'*/
+#Trimester Create (0.6ms)  INSERT INTO "trimesters" ("year", "term", "application_deadline", "start_date", "end_date", "created_at", "updated_at") VALUES ('2026', 'Spring', '2026-02-15', '2026-03-01', '2026-05-31', '2025-07-26 00:58:13.910265', '2025-07-26 00:58:13.910265') RETURNING "id" /*application='DocRails'*/
+#TRANSACTION (0.2ms)  COMMIT TRANSACTION /*application='DocRails'*/
+#<Trimester:0x000000010720c828
+ id: 9,
+ year: "2026",
+ term: "Spring",
+ application_deadline: "2026-02-15",
+ start_date: "2026-03-01",
+ end_date: "2026-05-31",
+ created_at: "2025-07-26 00:58:13.910265000 +0000",
+ updated_at: "2025-07-26 00:58:13.910265000 +0000">
+
+ spring_2026 = Trimester.find_by(year: '2026', term: 'Spring')
+
+ CodingClass.find_each do |coding_class|
+  puts "Creating new course for #{coding_class.title}"
+  Course.create!(
+    coding_class_id: coding_class.id,
+    trimester_id: spring_2026.id,
+    max_enrollment: 25
+  )
+  puts "Created course for #{coding_class.title}"
+ end
```

Very strong execution.
You used `find_each`, which is memory-efficient for larger datasets.
The puts statements demonstrate responsible debugging.
`create!` ensures that errors won’t silently fail. excellent practice!

Optional upgrade: use `find_or_create_by(...)` to avoid duplicates if the script runs multiple times.